backend/src/main/java/com/empire/World.java

```diff
@@ -1965,11 +1956,11 @@ void recruitTroops() {
 						getNation(k).gold -= signingBonus * recruits / 100;
 						incomeSources.getOrDefault(k, new Budget()).spentRecruits += signingBonus * recruits / 100;
 					}
-					List<String> tags = r.getArmyTags();
+					List<Army.Tag> tags = r.getArmyTags();
 					Army merge = null;
 					outer: for (Army a : armies) {
 						if (a.location == i && a.isArmy() && a.kingdom.equals(k)) {
-							for (String t : tags) if (!a.hasTag(t)) continue outer;
+							for (Army.Tag t : tags) if (!a.hasTag(t)) continue outer;
```

Comment:
Never knew you could do this outer thing in Java, you learn something new every day...

Reply:
Yeah, labels are not popular, but I do find them handy for this kind of thing.